### PR TITLE
Fix wrong placement of nodes created from dropped files. 

### LINKED ISSUE #5237: Fix drag-and-dropping a file
This task is automatically imported from the old Task Issue Board and it was originally created by Adam Obuchowicz.
Original issue is [here](https://www.pivotaltracker.com/story/show/184190708).
<hr />

The node is placed in the wrong place https://discord.com/channels/401396655599124480/1060952078069018684

and its expression is actually a non-working code.

## Comments:
The code placed has been fixed. 

The location of the dropped node is still wrong. (James Dunkerley - Feb 1, 2023)
<hr />

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Right now, we use the cursor position to determine the target position for dropped items. However, it seems that during dragging of files, we do not always receive mouse events, thus cannot update the cursor position. To avoid this, this PR refactors the functionality to use the location of the drop event, instead of the last known cursor position. 

Fixes #5237.

https://user-images.githubusercontent.com/1428930/224735951-9cd6ff62-a749-4ff3-8437-c0bee3c0dd05.mp4

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/view/graph-editor/src/lib.rs [hunk 1]
```diff
@@ -3749,12 +3749,12 @@ fn new_graph_editor(app: &Application) -> GraphEditor {
...
     frp::extend! { network
-        files_with_positions <- files_received.map3(&cursor_pos_in_scene,&default_gap,
-            |files,cursor_pos,default_gap| {
+        files_with_positions <- files_received.map2(&default_gap,
+            move |(drop_posititon,files),default_gap| {
                 let single_offset = default_gap + node::HEIGHT;
                 files.iter().enumerate().map(|(index,file)| {
                     let offset = Vector2(0.0, single_offset * index as f32);
-                    (file.clone_ref(),cursor_pos+offset)
+                    (file.clone_ref(),drop_posititon+offset)
```
Comment (farmaazon) [inline app/gui/view/graph-editor/src/lib.rs]: Add spaces after `,` and around `+`

File: lib/rust/ensogl/component/drop-manager/src/lib.rs [hunk 2]
```diff
@@ -112,6 +114,7 @@ impl File {
 
 type DropClosure = Closure<dyn Fn(web_sys::DragEvent)>;
 type DragOverClosure = Closure<dyn Fn(web_sys::DragEvent) -> bool>;
+type DropEventData = (Vector2, Vec<File>);
```
Comment (farmaazon) [inline lib/rust/ensogl/component/drop-manager/src/lib.rs]: I'd prefer  a structure with named fields.

File: lib/rust/ensogl/component/drop-manager/src/lib.rs [hunk 3]
```diff
@@ -122,26 +125,27 @@ type DragOverClosure = Closure<dyn Fn(web_sys::DragEvent) -> bool>;
 pub struct Manager {
     #[allow(dead_code)]
     network:          frp::Network,
-    files_received:   frp::Source<Vec<File>>,
+    files_received:   frp::Source<DropEventData>,
     #[allow(dead_code)]
     drop_handle:      web::EventListenerHandle,
     #[allow(dead_code)]
     drag_over_handle: web::EventListenerHandle,
+    scene:            Scene,
```
Comment (farmaazon) [inline lib/rust/ensogl/component/drop-manager/src/lib.rs]: We don't need to keep it as field, do we?

**Review by farmaazon — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]
❤️

**Comment by vitvakatu** [comment]
Unfortunately, it doesn't fix the issue I have on Mac. Perhaps it is important that the Enso window is unfocused at the moment of dropping.


https://user-images.githubusercontent.com/6566674/225261751-526a6ee1-63f8-4a1c-8d61-ef3f5f604257.mp4

**Comment by wdanilo** [comment]
@vitvakatu you're so good at doing Q/A! thanks so much for checking it always so well :)

**Review by vitvakatu — APPROVED** [review]
QA accepted

**Pull request merged**